src/java.base/share/classes/java/lang/System.java

```diff
@@ -124,10 +124,19 @@ private System() {
      *
      * @apiNote
      * The typical approach to read character data is to wrap {@code System.in}
-     * within an {@link java.io.InputStreamReader InputStreamReader} or other object
-     * that handles character encoding. After this is done, subsequent reading should
-     * use only the wrapper object; operating directly on {@code System.in} results
-     * in unspecified behavior.
+     * within the object that handles character encoding. After this is done,
+     * subsequent reading should use only the wrapper object; operating directly
+     * on {@code System.in} results in unspecified behavior.
+     * <p>
+     * Here are two common examples. Using an {@link java.io.InputStreamReader
+     * InputStreamReader}:
+     * {@snippet lang=java :
+     *     new InputStreamReader(System.in, System.getProperty("stdin.encoding"));
+     * }
```

Comment:
So, ok as far as it goes.  Does this example imply that one should always use the 2 arg constructor.
When should the 1-arg constructor that uses the defaultEncoding vs the 2-arg constructor with stdin.encoding be used?
Is that understood to be described elsewhere?

Reply:
For reading from `System.in`, yes. For other cases, depends on the user's situation. This is the first to modify our documents to guide users as `stdin.encoding` is a new property. There are some descriptions for `System.out` and `System.err` in the JEP 400 though. Probably we will need to update the JEP too.